src/Hedgehog/Property.fs

```diff
@@ -81,11 +81,29 @@ module Property =
     let bind (k : 'a -> Property<'b>) (m : Property<'a>) : Property<'b> =
         bindGen (toGen << k) (toGen m) |> ofGen
 
+
+    // sprintf "%A" is not prepared for printing ResizeArray<_> (C# List<T>) so we prepare the value instead
+    let private prepareForPrinting (value: obj) : obj =
+#if FABLE_COMPILER
+        value
+#else
+        let t = value.GetType()
+        // have to use TypeInfo due to targeting netstandard 1.6
+        let t = System.Reflection.IntrospectionExtensions.GetTypeInfo(t)
+        let isList = t.IsGenericType && t.GetGenericTypeDefinition() = typedefof<ResizeArray<_>>
+        if isList
+        then value :?> System.Collections.IEnumerable |> Seq.cast<obj> |> List.ofSeq :> obj
+        else value
+#endif
+
+    let private printValue (value) : string =
+        value |> prepareForPrinting |> sprintf "%A"
```

Comment:
I think this can be inlined

Reply:
It can, but isn't it better to let the compiler do the inlining? This way we know the intent of the code, captured in a name, and are ready to reuse it should the need arise.

Reply:
Ok, good point.